### PR TITLE
migrates to maven 3 and java 8, ignores tests that were already faili…

### PR DESCRIPTION
**Pull request by billy-mccarthy**
upgrades to use maven 3 and java 8 
bumps dom4j version to hash release with security fix 
ignores tests that were failing on checkout 

tested that dom4j builds with this by building the snapshot version and running the dom4j test suite. All tests passed

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  <description>Jaxen is a universal Java XPath engine.</description>
+  <url>http://jaxen.codehaus.org/</url>
+  <inceptionYear>2001</inceptionYear>
+  <organization>
+    <name>Codehaus</name>
+    <url>http://www.codehaus.org</url>
+  </organization>
+
+  <scm>
+    <connection>scm:cvs:pserver:anonymous@cvs.jaxen.codehaus.org:/home/projects/jaxen/scm:jaxen</connection>
```
Comment (andrewswan) [inline pom.xml]: These don't look like GitHub SCM details.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          <moduleExcludes>
+            <xdoc>navigation.xml</xdoc>
+          </moduleExcludes>
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <excludes>
+            <exclude>org/jaxen/jdom/XPathTest.java</exclude>
```
Comment (andrewswan) [inline pom.xml:116]: How about an XML comment to explain this exclusion?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    </plugins>
+  </build>
+
+  <reporting>
+    <plugins>
+      <plugin>
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <configuration>
+          <links>
+            <link>http://java.sun.com/j2se/1.4.2/docs/api/index.html</link>
```
Comment (andrewswan) [inline pom.xml]: Seems like an old version for a project that uses Java 8.

**Review by andrewswan — APPROVED** [review]
I've reviewed the POM, which is probably what you've spent the most time on. I haven't looked so closely at the Java code, as there's so much of it. Let me and any other reviewers know if there's some part of the code that needs special attention.

File: pom.xml (hunk 1 above)
Comment (billy-mccarthy) [reply pom.xml]: yep sorry these were generated, I'll update the and the distribution management info

File: src/main/java/org/jaxen/dom/NamespaceNode.java [hunk 4]
```diff
@@ -480,6 +481,66 @@ public boolean hasAttributes ()
         return false;
     }
 
+    @Override
```
Comment (billy-mccarthy) [inline src/main/java/org/jaxen/dom/NamespaceNode.java:484]: These were required to be implemented as part of the Java 8 upgrade

File: src/test/java/org/jaxen/dom4j/XPathTest.java [hunk 5]
```diff
@@ -152,7 +151,7 @@ public void testJaxen20AttributeNamespaceNodes() throws JaxenException
         Namespace ns1 = Namespace.get("p1", "www.acme1.org");
         Namespace ns2 = Namespace.get("p2", "www.acme2.org");
         Element element = new DefaultElement("test", ns1);
-        Attribute attribute = new DefaultAttribute("pre:foo", "bar", ns2);
+        Attribute attribute = new DefaultAttribute("prefoo", "bar", ns2);
```
Comment (billy-mccarthy) [inline src/test/java/org/jaxen/dom4j/XPathTest.java:154]: dom4j fix does not allow the character :

File: src/test/java/org/jaxen/jdom/XPathTest.java [hunk 6]
```diff
@@ -162,7 +161,7 @@ public void testJaxen53Text() throws JaxenException, JDOMException, IOException
 
     }
     
-    public void testJaxen20AttributeNamespaceNodes() throws JaxenException
+    public void ignoreTestJaxen20AttributeNamespaceNodes() throws JaxenException
```
Comment (billy-mccarthy) [inline src/test/java/org/jaxen/jdom/XPathTest.java:164]: these tests were failing on checkout of the code before bumping the dom4j version

File: pom.xml (hunk 2 above)
Comment (billy-mccarthy) [reply pom.xml:116]: This was generated by mvn one:convert I'm not sure why it exists but will see if I can find out

**Comment by billy-mccarthy** [comment]
See https://github.com/atlassian/jaxen/pull/2 for same PR with different branch name

**Pull request closed without merging**